lib/from_openstudio/material/opaque_no_mass.rb

```diff
@@ -42,18 +42,51 @@ def self.from_material(material)
         # set hash values from OpenStudio Object
         hash[:identifier] = material.nameString
         hash[:r_value] = material.thermalResistance
-        hash[:roughness] = material.roughness
-        # check if boost optional object is empty
-        unless material.thermalAbsorptance.empty?
-          hash[:thermal_absorptance] = material.thermalAbsorptance.get
+        # Roughness is a required property for OS MasslessOpaqueMaterial but isn't a listed property
+        # for OS AirGap
+        begin
+          hash[:roughness] = material.roughness
+        rescue
         end
-        # check if boost optional object is empty
-        unless material.solarAbsorptance.empty?
-          hash[:solar_absorptance] = material.solarAbsorptance.get
+        if hash[:roughness].nil?
+          # If property doesn't exist, use default value from schema
+          hash[:roughness] = defaults[:roughness][:default]
         end
         # check if boost optional object is empty
-        unless material.visibleAbsorptance.empty?
-          hash[:visible_absorptance] = material.visibleAbsorptance.get
+        begin
+          # check if boost optional object is empty
+          if !material.thermal_absorptance.empty? hash[:thermal_absorptance] = material.thermalAbsorptance.get
+          end
+        rescue
+        end
+        
+        # if property doesn't exist use default value from schema
+        if hash[:thermal_absorptance].nil?
+          hash[:thermal_absorptance] = defaults[:thermal_absorptance][:default]
```

Comment:
This is adding defaults from the schema for MasslessOpaqueMaterial material if properties arent defined as well as AirGap

Reply:
We can just get rid of this part. [These fields are all optional](https://www.ladybug.tools/honeybee-schema/model.html#tag/energymaterial_model) and the core Python libraries automatically fill in the default when they aren't specified in the JSON. In fact, I think this might be a case where "less is more" and we should just keep the total amount of code we need to maintain to a minimum.